lib/registries/addon/drafts/draft/route.ts

```diff
@@ -0,0 +1,46 @@
+import { action } from '@ember-decorators/object';
+import { service } from '@ember-decorators/service';
+import Route from '@ember/routing/route';
+import RouterService from '@ember/routing/router-service';
+import { task } from 'ember-concurrency';
+import DS from 'ember-data';
+
+import requireAuth from 'ember-osf-web/decorators/require-auth';
+import Analytics from 'ember-osf-web/services/analytics';
+
+@requireAuth()
+export default class DraftRegistrationRoute extends Route.extend({
+    loadModelTask: task(function *(this: DraftRegistrationRoute, draftId: string) {
+        try {
+            return yield this.store.findRecord('draft-registration', draftId);
+        } catch (error) {
+            this.transitionTo('not-found', this.router.currentURL.slice(1));
+            return undefined;
+        }
+    }),
+}) {
+    @service analytics!: Analytics;
+    @service store!: DS.Store;
+    @service router!: RouterService;
+
+    model(this: DraftRegistrationRoute, params: { id: string, page?: string }) {
+        const { id, page } = params;
+
+        if (!page) {
+            this.transitionTo('drafts.draft', id, '1-');
+        }
+
+        const match = page ? page.match(/^\d+(?=-)/) : undefined;
```

Comment:
Since we'll be parsing/building the `page` param in multiple places, how about let's put that logic in its own file (maybe `drafts/draft/page-param.ts`?), with exports like:
- `getPageIndex(pageParam: string): number`
  - e.g. `getPageIndex('1-foo') === 0`
- `getPageParam(pageIndex: number, pageHeading?: string): string`
  - e.g. `getPageParam(0, 'bar') === '1-bar'`
- `getDefaultPageParam(): string`
  - returns `'1'`
  - could also be an exported `const`

Reply:
Makes sense. Shouldn't `page-param` be a util instead?

Reply:
It could be! Though I think of utils as things that we expect will be useful in multiple situations, and this seems very specific to these two routes. I don't have strong feelings either way 🤷‍♂